doc/USP0017.md

```diff
@@ -0,0 +1,31 @@
+# USP0017 Unity objects should not use coalescing assignment
+
+`UnityEngine.Object` should not be used with the coalescing assignment operator.
+
+## Suppressed Diagnostic ID
+
+IDE0074 - Use compound assignment
+
+## Examples of code that produces a suppressed diagnostic
+```csharp
+using UnityEngine;
+
+class Camera : MonoBehaviour
+{
+	public Transform a;
+	public Transform b;
+
+	public Transform NC()
+	{
+		return a ?? (a = b);
```

Comment:
I think we also need to say in the doc that a ?? (a = b) should not be used with UE.Objects either and that form will result in us raising a diagnostic.